src/main/java/io/airlift/testing/mysql/EmbeddedMySql.java

```diff
@@ -0,0 +1,295 @@
+
+            mysqld.destroyForcibly();
+
+            try {
+                mysqld.waitFor(SHUTDOWN_WAIT.toMillis(), MILLISECONDS);
+            }
+            catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+
+            if (mysqld.isAlive()) {
+                log.error("mysqld is still running in %s", serverDirectory);
+            }
+        }
+
+        try {
+            deleteRecursively(serverDirectory, ALLOW_INSECURE);
+        }
+        catch (IOException e) {
+            log.warn(e, "Failed to delete %s", serverDirectory);
+        }
+
+        executor.shutdownNow();
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .add("serverDirectory", serverDirectory)
+                .add("port", port)
+                .toString();
+    }
+
+    private static int randomPort()
+            throws IOException
+    {
+        try (ServerSocket socket = new ServerSocket(0)) {
+            return socket.getLocalPort();
+        }
+    }
+
+    private void initialize()
+    {
+        system(mysqld(),
+                "--no-defaults",
+                "--initialize-insecure",
+                "--datadir", dataDir());
+    }
+
+    private Process startMysqld()
+            throws IOException
+    {
+        List<String> args = newArrayList(
+                mysqld(),
+                "--no-defaults",
+                "--skip-ssl",
+                "--disable-partition-engine-check",
+                "--explicit_defaults_for_timestamp",
+                "--lc_messages_dir", serverDirectory.resolve("share").toString(),
+                "--socket", serverDirectory.resolve("mysql.sock").toString(),
+                "--port", String.valueOf(port),
+                "--datadir", dataDir());
+
+        Process process = new ProcessBuilder(args)
+                .redirectErrorStream(true)
+                .redirectOutput(Redirect.INHERIT)
+                .start();
+
+        log.info("mysqld started on port %s. Waiting up to %s for startup to finish.", port, STARTUP_WAIT);
+
+        waitForServerStartup(process);
+
+        return process;
+    }
+
+    private String mysqld()
+    {
+        return serverDirectory.resolve("bin").resolve("mysqld").toString();
+    }
+
+    private String dataDir()
+    {
+        return serverDirectory.resolve("data").toString();
+    }
+
+    private void waitForServerStartup(Process process)
+            throws IOException
+    {
+        Throwable lastCause = null;
+        long start = System.nanoTime();
+        while (Duration.nanosSince(start).compareTo(STARTUP_WAIT) <= 0) {
+            try {
+                checkReady();
+                log.debug("mysqld startup finished");
+                return;
+            }
+            catch (SQLException e) {
+                lastCause = e;
+                log.debug("Waiting for mysqld startup", e);
```

Comment:
This could print every 10ms

Reply:
I'll just remove this. It's always going to be something like "connection refused" which we expect.